driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -1938,7 +1942,13 @@ private static ClientReplaceOneOptions getClientReplaceOneOptions(final BsonDocu
     }
 
     private static ClientUpdateOneOptions getClientUpdateOneOptions(final BsonDocument arguments) {
-        return fillAbstractClientUpdateOptions(new ConcreteClientUpdateOneOptions(), arguments);
+        ConcreteClientUpdateOneOptions options = new ConcreteClientUpdateOneOptions();
+
+        ofNullable(arguments.remove("sort"))
+                .map(BsonValue::asDocument)
+                .ifPresent(options::sort);
```

Comment:
I think we should avoid using Optional. Return values and the occasional "orElse" seems fine, but it becomes quite confusing when chaining operations are used - even in this small example, I think the following is much clearer:
```suggestion
        if (arguments.containsKey("sort")) {
            options.sort(arguments.get("sort").asDocument());
        }
```
This does not use "remove" - I think that was not intended, since it modifies arguments?

Reply:
I see your point, and I used Optional here for conciseness, as I thought it kept the example readable. I noticed we use method chaining with streams and Optional in other parts of the codebase, so it might be worth discussing this as a broader code style topic. However, I’ve changed it to use an `if - else` for now.

Regarding `arguments.remove`, it is used because `fillAbstractClientUpdateOptions` is a shared method between `UpdateOneOptions` and `UpdateManyOptions`, and it throws an exception for unknown options. Since `sort` is not a shared option, it is removed to allow processing. We use a similar pattern here:
[UnifiedCrudHelper example](https://github.com/mongodb/mongo-java-driver/blob/01eaf33e1d34e1e5c3e32d54fed3d23f5df0db99/driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java#L1756).